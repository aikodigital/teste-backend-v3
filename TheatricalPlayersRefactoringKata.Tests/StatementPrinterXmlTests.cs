using Xunit;
using ApprovalTests;
using ApprovalTests.Reporters;
using System.Collections.Generic;
using TheatricalPlayersRefactoringKata;

public class StatementPrinterXmlTests
{
    [Fact]
    [UseReporter(typeof(DiffReporter))]
    public void TestXmlStatementWithHistoryPlay()
    {
        var plays = new Dictionary<string, Play>
        {
            { "henry-v", new Play("Henry V", 3227, "history") },
            { "john", new Play("King John", 2648, "history") },
            { "richard-iii", new Play("Richard III", 3718, "history") }
        };

        var invoice = new Invoice("BigCo", new List<Performance>
        {
            new Performance("henry-v", 20),
            new Performance("john", 39)
        });

        var statementPrinter = new StatementPrinter();

        var result = statementPrinter.PrintXml(invoice, plays);

        Approvals.Verify(result);
    }
}
