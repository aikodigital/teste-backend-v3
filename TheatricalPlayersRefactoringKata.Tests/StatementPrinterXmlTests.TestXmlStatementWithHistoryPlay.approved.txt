<?xml version="1.0" encoding="utf-16"?>
<statement>
  <customer>BigCo</customer>
  <performance play="Henry V" amount="¤705.40" audience="20" />
  <performance play="King John" amount="¤931.60" audience="39" />
  <totalAmount>¤1,637.00</totalAmount>
  <volumeCredits>9</volumeCredits>
</statement>